src/plugins/network.ts

```diff
@@ -23,336 +23,217 @@
+      isNavigationRequest: request.isNavigationRequest(),
+      status: -1,
       loadEndTime: -1,
       responseReceivedTime: -1,
-      response: null,
-      resourceSize: 0,
-      transferSize: 0,
+      resourceSize: -1,
+      transferSize: -1,
       timings: null,
-    });
-  }
-
-  _onRequestWillBeSentExtraInfo(
-    event: Protocol.Network.requestWillBeSentExtraInfoPayload
-  ) {
-    const { requestId, headers } = event;
-    const record = this._findNetworkRecord(requestId);
-    if (!record) {
-      return;
-    }
-    /**
-     * Enhance request headers with additional information
-     */
-    record.request.headers = {
-      ...record.request.headers,
-      ...headers,
     };
-  }
 
-  _onDataReceived(event: Protocol.Network.dataReceivedPayload) {
-    const { requestId, dataLength, encodedDataLength } = event;
-    const record = this._findNetworkRecord(requestId);
-    if (!record) {
-      return;
+    if (request.redirectedFrom()) {
+      const fromEntry = this._findNetworkEntry(request.redirectedFrom());
+      if (fromEntry) fromEntry.response.redirectURL = request.url();
     }
-    record.resourceSize += dataLength;
-    if (encodedDataLength >= 0) record.transferSize += encodedDataLength;
-  }
-
-  _onResponseReceived(event: Protocol.Network.responseReceivedPayload) {
-    const { requestId, response, timestamp } = event;
-    const record = this._findNetworkRecord(requestId);
-    if (!record) {
-      return;
+    (request as any)[this._networkEntrySymbol] = networkEntry;
+    this.results.push(networkEntry);
+  }
+
+  private async _onResponse(response: Response) {
+    const request = response.request();
+    const networkEntry = this._findNetworkEntry(request);
+    if (!networkEntry) return;
+
+    const server = await response.serverAddr();
+    const responseHeaders = await response.allHeaders();
+    const mimeType = responseHeaders['content-type']
+      ? responseHeaders['content-type'].split(';')[0]
+      : 'unknown';
+
+    const requestHeaders = await request.allHeaders();
+
+    networkEntry.request.headers = requestHeaders;
+    networkEntry.request.referrer = requestHeaders?.referer;
+    networkEntry.response = {
+      url: response.url(),
+      statusCode: response.status(),
+      statusText: response.statusText(),
+      headers: responseHeaders,
+      mimeType,
+      redirectURL: networkEntry.response.redirectURL,
+      securityDetails: await response.securityDetails(),
+      remoteIPAddress: server?.ipAddress,
+      remotePort: server?.port,
+    };
+    networkEntry.status = response.status();
+    networkEntry.responseReceivedTime = epochTimeInMills();
+  }
+
+  private async _onRequestCompleted(request: Request) {
+    const networkEntry = this._findNetworkEntry(request);
+    if (!networkEntry) return;
+
+    networkEntry.loadEndTime = epochTimeInMills();
+
+    // For aborted/failed requests sizes does not exist
+    const sizes = await request.sizes().catch(() => {});
```

Comment:
Could we add an `isFailedRequest` boolean to the NetworkInfo type to more easily identify failed/aborted requests on the front end?

Reply:
`networkEntry.response` would be null for failed reqs.